docs/usage.rst

```diff
@@ -26,44 +26,103 @@ configured storage nodes.  The client nodes must all share the same
 storage nodes.  The nodes must be running.
 
 
-Running a Magic Folder process
-------------------------------
+Creating a Magic Folder Daemon Configuration
+--------------------------------------------
+
+The "Magic Folder Daemon" is a long-running process that accomplishes
+the task of synchronizing all the "magic folders" that are configured
+in it.
+
+A Magic Folder Daemon needs to have some configuration to start. There
+are two ways to create this: from scratch; or from a "legacy"
+magic-folder (when ``magic-folder`` was a sub-command of ``tahoe`` in
+version 1.14.0 and earlier).
+
+No matter the method used, there are some pieces of information
+required and two decisions to be made. You need to know:
+
+- the ``node-directory`` of the Tahoe-LAFS client this magic folder daemon shall use.
+
+You need to decide:
+
+- The network endpoint the API will listen on. This is how CLI commands and front-ends communciate to the magic-folder daemon and is expressed as a Twisted "server endpoint string". For example, ``tcp:4321`` to listen globally on port 4321. You may also use Unix Domain sockets on OSes which support that; for example ``unix:/var/run/magic-folder/api-socket``.
```

Comment:
yes, I agree unix would be ideal .. I was going to write that down in here but then didn't know what HTTP clients like treq would make of that so went with ... something more familiar ;)

I think also `onion:80` would work too (if you installed `txtorcon` first) .. that at least actually-is TCP under the hood so web clients have a way higher chance of it working. See also my comment on https://github.com/LeastAuthority/magic-folder/issues/222#issuecomment-668291181 in this direction...

I'm thinking out loud right now that *maybe* a way around this is to force the USER to specify how to contact it (or, to do so optionally or in "weird" circumstances like "unix sockets" or "tor" or "i2p"). That is, there is simply config for both the server-side and client-side (the latter maybe expressed as a URI?) and for the simple case (TCP) we can auto-fill the client-side one.

Thinking: if you say `unix:/foo/bar` for the server you have to say `http:///foo/bar` or whatever treq might accept for the client? (Or, for one which stands a chance of working: `onion:80` on the server and `http://iasdfiasdfoijasdofijaosdfijasodfijaodfsijaosdifjasdf.onion` for the client-side, approximately). This would also be useful for people doing somewhat-standard port-forwarding setups, like you say `tcp:5000:interface=192.168.2.1` or something for the server but `https://example.com/` for the client. I'm probably drastically over-thinking the latter cases, maybe, since they'd need to access the token as well so really they do need to be on the same machine......but also there's things like NFS and shared drives etc so :shrug:

Reply:
`twm` in `#twisted` says that treq can use unix (or, I guess, "whatever") style endpoints with `Agent.usingEndpointFactory` and it will then ignore the host-part of the URIs you feed it.